tests/lib/azure/alb.go

```diff
@@ -0,0 +1,235 @@
+	"fmt"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/NeowayLabs/klb/tests/lib/azure/fixture"
+)
+
+type LoadBalancers struct {
+	client network.LoadBalancersClient
+	f      fixture.F
+}
+
+type LoadBalancerProbe struct {
+	Name     string
+	Protocol string
+	Port     int32
+	Interval int32
+	Count    int32
+	Path     string
+}
+
+func NewLoadBalancers(f fixture.F) *LoadBalancers {
+	as := &LoadBalancers{
+		client: network.NewLoadBalancersClient(f.Session.SubscriptionID),
+		f:      f,
+	}
+	as.client.Authorizer = f.Session.Token
+	return as
+}
+
+// AssertExists checks if load balancer exists in the resource group.
+// Fail tests otherwise.
+func (lb *LoadBalancers) AssertExists(
+	t *testing.T,
+	name string,
+	frontendipName string,
+	privateIP string,
+	poolname string,
+) {
+	lb.f.Retrier.Run(newID("LoadBalancers", "AssertExists", name), func() error {
+		loadbalancer, err := lb.getLoadBalancer(t, name)
+		if err != nil {
+			return err
+		}
+		return assertConfig(t, loadbalancer, frontendipName, privateIP, poolname)
+	})
+}
+
+// AssertProbeExists checks if load balancer exists and it has the given probe.
+// Fail tests otherwise.
+func (lb *LoadBalancers) AssertProbeExists(t *testing.T, lbname string, p LoadBalancerProbe) {
+	lb.f.Retrier.Run(newID("LoadBalancers", "AssertProbeExists", p.Name), func() error {
+		loadbalancer, err := lb.getLoadBalancer(t, lbname)
+		if err != nil {
+			return err
+		}
+		prop := loadbalancer.LoadBalancerPropertiesFormat
+		if prop == nil {
+			return fmt.Errorf("no properties found on: %s", loadbalancer)
+		}
+		if prop.Probes == nil {
+			return fmt.Errorf("no probes found on: %s", prop)
+		}
+
+		for _, probe := range *prop.Probes {
+			if probe.Name == nil {
+				continue
+			}
+			name := *probe.Name
+			lb.f.Logger.Printf("analyzing probe: %q", name)
+			if name != p.Name {
+				continue
+			}
+			probeProperties, err := checkProbePropertiesFormat(probe.ProbePropertiesFormat)
+			if err != nil {
+				return err
+			}
+			protocol := string(probeProperties.Protocol)
+			port := *probeProperties.Port
+			path := *probeProperties.RequestPath
+			interval := *probeProperties.IntervalInSeconds
+			if string(probeProperties.Protocol) != p.Protocol {
+				return fmt.Errorf(
+					"expected probe protocol: %q, got: %q",
+					p.Protocol,
+					protocol,
+				)
+			}
+			if port != p.Port {
+				return fmt.Errorf(
+					"expected probe port: %d, got: %d",
+					p.Port,
+					port,
+				)
+			}
+			if interval != p.Interval {
+				return fmt.Errorf(
+					"expected probe interval: %d, interval: %d",
+					p.Interval,
+					interval,
+				)
+			}
+			if path != p.Path {
+				return fmt.Errorf(
+					"expected probe port: %s, got: %s",
+					p.Path,
+					path,
+				)
+			}
+			return nil
+		}
+		return fmt.Errorf("unable to find probe: %+v on lb: %s", p, lbname)
+	})
+}
+
+func checkProbePropertiesFormat(p *network.ProbePropertiesFormat) (*network.ProbePropertiesFormat, error) {
```

Comment:
@tiago4orion 

This kind of validation/crappy code could be solver without all this code using reflection. This is the third time I have to do this and it pretty annoying :-(

It is a very specific case of an API that is filled with structs where everything is pointers and it is just hell to validate.

One of the ideas I got was something like:

```
equal.IsSubset(data1, data2)
```

It would use reflection to check if all fields on struct data1 is present on data2 (I would check for a subset because on the tests we ignore some fields).

The reflection code would do all the heavy lifting navigating the struct and checking for nil crap.

Not sure if it is a great idea...but we must have a lot of these crappy/dump validation code hanging around...and I'm getting pissed about writing this kind of shit :-(